public/static/app/perf/trend_chart.js

```diff
@@ -10,124 +14,120 @@ mciModule.factory('DrawPerfTrendChart', function (
       })
     }
 
     function hydrateChangePoint(point) {
       // if idx == 0, shift it right for one point
-      var revIdx = idxByRevision(series, point.suspect_revision) || 1
+      const revIdx = idxByRevision(series, point.suspect_revision) || 1;
 
       point._meta = {
         firstRevIdx: revIdx - 1,
-        lastRevIdx:  revIdx,
+        lastRevIdx: revIdx,
       }
     }
 
     // Filter out change points which lays outside ot the chart
     let visibleChangePoints = (metric === "ops_per_sec") ? _.chain(changePoints)
-      .filter(function(d) {
+      .filter(function (d) {
         return _.findWhere(series, {revision: d.suspect_revision})
       })
       .each(hydrateChangePoint) // Add some useful meta data
       .value() : [];
 
-    var visibleBFs = _.filter(buildFailures, function(d) {
+    const visibleBFs = _.filter(buildFailures, function (d) {
       return _.findWhere(series, {revision: d.first_failing_revision})
-    })
+    });
 
-    var cfg = PerfChartService.cfg
+    const cfg = PerfChartService.cfg;
 
     // override default metric
-    cfg.valueAttr = metric
+    cfg.valueAttr = metric;
 
     document.getElementById(containerId).innerHTML = '';
 
-    var svg = d3.select('[id="' + containerId + '"]')
+    const svg = d3.select('[id="' + containerId + '"]')
       .append('svg')
       .attr({
         class: 'series',
         width: cfg.container.width,
         height: cfg.container.height,
-      })
+      });
 
-    var colors = d3.scale.category10();
+    const colors = d3.scale.category10();
     // FIXME Force color range 'initialization'. Might be a bug in d3 3.5.3
     // For some reason, d3 will not give you, let's say, the third color
     // unless you requested 0, 1 and 2 before.
     for (var i = 0; i < cfg.knownLevelsCount; i++) colors(i);
-
-    var ops = _.pluck(series, cfg.valueAttr);
-    var opsValues = _.pluck(series, cfg.valueAttr + "_values");
+    _.pluck(series, cfg.valueAttr);
+    const opsValues = _.pluck(series, cfg.valueAttr + "_values");
 
     // Currently selcted revision item index
-    var currentItemIdx = _.findIndex(series, function(d) {
-      return d.task_id == scope.task.id
-    })
+    const currentItemIdx = _.findIndex(series, function (d) {
+      return d.task_id === scope.task.id
+    });
 
-    function extractThreadLevels(series) {
-      return _.reduce(series, function(m, d) {
-        return _.union(
-          m, _.pluck(d.threadResults, 'threadLevel')
-        )
-      }, [])
+    const allLevelsByIndex = series.map(entry => entry.threadResults.map(result => result.threadLevel));
+    const allLevels = allLevelsByIndex.reduce((oneLevel, anotherLevel) => _.union(oneLevel, anotherLevel));
+    const maxLevelByIndex = allLevelsByIndex.map(level => _.max(level));
+
+    function maxLevelIdx(seriesIndex) {
+      return series[seriesIndex].threadResults.map(result => result.threadLevel).indexOf(maxLevelByIndex[seriesIndex]);
```

Comment:
This seems to be selecting the index of the largest thread level rather than the index of the largest result.

Reply:
:+1: